java/hcx-integrator-sdk/src/main/java/io/hcxprotocol/validator/HCXFHIRValidator.java

```diff
@@ -24,9 +33,36 @@ public class HCXFHIRValidator {
     private HCXFHIRValidator(String hcxIGBasePath, String nrcesIGBasePath) throws Exception {
         FhirContext fhirContext = FhirContext.forR4();
         fhirContext.setDefaultTypeForProfile(hcxIGBasePath + "StructureDefinition-HCXInsurancePlan.html", HCXInsurancePlan.class);
-        // Create a chain that will hold the validation modules
+        //   Create a chain that will hold the validation modules
         System.out.println("we have started");
         ValidationSupportChain supportChain = new ValidationSupportChain();
+        String currentDir = System.getProperty("user.dir") + "\\";
+
+        Path newDirNRCES = Paths.get(currentDir, "nrces_definitions");
+        Path newDirHCX = Paths.get(currentDir, "hcx_definitions");
+
+        File nrcesFile = new File(newDirNRCES.toString());
+        boolean isPresentNRCESPath = nrcesFile.exists() && nrcesFile.isDirectory();
+        if (!isPresentNRCESPath) {
+            Files.createDirectory(newDirNRCES);
+        }
+
+        File hcxFile = new File(newDirHCX.toString());
+        boolean isPresentHCXPath = hcxFile.exists() && hcxFile.isDirectory();
+        if (!isPresentHCXPath) {
+            Files.createDirectory(newDirHCX);
+        }
+
+        downloadZip(new URL(hcxIGBasePath), currentDir + "hcx_definitions.zip");
+        downloadZip(new URL(nrcesIGBasePath), currentDir + "nrces_definitions.zip");
+
+       // extract zip files
+        if (!isPresentNRCESPath) {
+            decompressZIP(Paths.get(currentDir, "nrces_definitions.zip"), newDirNRCES);
+        }
+        if (!isPresentHCXPath) {
+            decompressZIP(Paths.get(currentDir, "hcx_definitions.zip"), newDirHCX);
+        }
```

Comment:
Download profiles is same for HCX and NRCES, Generalize the code to use common method.